### PR TITLE
Issue62 task get route for all existing listings

### LINKED ISSUE #62: [TASK] Implement a GET route for retrieving all existing listings from DB
### Associated User Story: 
#23 or/and #26 

### Task Description:
- This ticket aims to implement an API endpoint for front-end to retrieve all existing listings from DB.
- Route: `/api/v1/listing`
- Areas that involve updates:
   - `/api` folder (**required**)
   - `/controllers` folder (**required**)
   - `/service` folder (optional - TBD - may have another task ticket to refactor codes from this)
   - `/dao` folder (optional - TBD - may have another task ticket to refactor codes from this)
   - `/dto` folder (optional - TBD - may have another task ticket to refactor codes from this)

### Dependencies:
N/A

### Acceptance Criteria (may be updated in the future):
- `/api/v1/listing` returns a list of JSON objects, and each object should consist of fields as follows:
   1. Id,
   2. Title, 
   3. Price, 
   4. Number of bedrooms, 
   5. Number of bathrooms, and 
   6. Description.

### PR DESCRIPTION
**Pull request by AzizulHakim96**
### Associated issues:
User story: #26 
dev task: #62 

### Task Description:
This ticket aims to implement an API endpoint for the front-end to retrieve all existing listings from DB.

testPassed
GetAllListings

### Github action
Closes #62

### CODE REVIEW

**Review by jcsnorlax97 — CHANGES_REQUESTED** [review]
It's good in general. Point out some inconsistent places below. Would approve it after the changes have been made. (make the changes in the same branch, `issue62-XXX`, and push it to github; then it will be included inside this PR automatically.)

**Review by Mik3Qiao — APPROVED** [review]

File: server/controllers/listing.js [hunk 1]
```diff
@@ -16,6 +16,20 @@ class ListingController {
...
   };
+
+
+  getAllListings = async (req, res, next) => {
+    try {
+      const listing = await this.listingService.getAllListings(req.params);
+      if (listing == null) {
+        next(ApiError.notFound(`No listing found!`));
+        return;
+      }
```
[the remark this answers is not in the document]
Comment (AzizulHakim96) [reply server/controllers/listing.js:27]: I think if I just remove the if-case, everything will work fine. Let me know if you think it is necessary or not to make an array in /dao/listing.js

[the remark this answers is not in the document]
Comment (jcsnorlax97) [reply server/controllers/listing.js:27]: Yeah. We can remove that if-case. Let me know when you have removed it and pushed the change. I will approve this after that.

[the remark this answers is not in the document]
Comment (AzizulHakim96) [reply server/controllers/listing.js]: Updated!

File: server/controllers/listing.js [hunk 2]
```diff
@@ -16,6 +16,16 @@ class ListingController {
       next(ApiError.internal(`${err}`));
     }
   };
+
+  getAllListings = async (req, res, next) => {
+    try {
+      const listing = await this.listingService.getAllListings(req.params);
+
```
Comment (jcsnorlax97) [inline server/controllers/listing.js:23]: Super super super picky: just point this out, you don't need to remove this extra empty line lol

**Review by jcsnorlax97 — APPROVED** [review]
Approved! Great Job, @AzizulHakim96!

File: server/controllers/listing.js (hunk 2 above)
Comment (AzizulHakim96) [reply server/controllers/listing.js:23]: All good

**Review by RonnieDsouza — APPROVED** [review]

**Comment by RonnieDsouza** [comment]
Looks Good Aziz!

**Pull request merged**